pyproject.toml

```diff
@@ -1,3 +1,19 @@
+[project]
+requires-python = ">=3.10"
```

Comment:
add a `<` selector too, to prevent this from eagerly selecting a version of python that we don't currently support

Reply:
ah, good point, I mostly added it because it also serves as a hint to ruff to guide it in some of the linters.

Any preferences on how strict we want the requirement to be? [Django's supported versions](https://docs.djangoproject.com/en/5.2/faq/install/#what-python-version-can-i-use-with-django), i.e. `<3.13`? `<4.0`?